e2e/steps/drag-column-basic.steps.ts

```diff
@@ -3,15 +3,16 @@ import { element, by, browser } from 'protractor';
 
 When('I drag column {string} to column {string}', (elem, target) => dragElement(elem, target));
 
-// DND doesn't work now
 async function dragElement(elem, target) {
-	const el = element(by.xpath(`//*[contains(text(),'" + elem + "')]/../..`));
-	const trg = element(by.xpath(`//*[contains(text(),'" + target + "')]/../..`));
-	browser.actions()
-		.mouseMove(el.getWebElement(), { x: 0, y: 0 })
-		.mouseDown()
-		.mouseMove(el.getWebElement(), { x: 20, y: 20 })
-		.mouseMove(trg.getWebElement(), { x: 5, y: 5 })
-		.mouseUp()
-		.perform();
+	await browser.sleep(1000);
+	const el = element(by.xpath(`//*[contains(text(),'${elem}')]/../..`));
+	const trg = element(by.xpath(`//*[contains(text(),'${target}')]/../..`));
+
+	doDnd(el, trg);
 }
+
+function doDnd(dragElement, dropElement) {
+    const CUSTOM_JS_SCRIPT = 'function e(e,t,n,i){var r=a.createEvent("DragEvent");r.initMouseEvent(t,!0,!0,o,0,0,0,c,g,!1,!1,!1,!1,0,null),Object.defineProperty(r,"dataTransfer",{get:function(){return d}}),e.dispatchEvent(r),o.setTimeout(i,n)}var t=arguments[0],n=arguments[1],i=arguments[2]||0,r=arguments[3]||0;var a=t.ownerDocument,o=a.defaultView,l=t.getBoundingClientRect(),u=n?n.getBoundingClientRect():l,c=l.left+(l.width>>1),g=l.top+(l.height>>1),s=u.left+(u.width>>1)+i,f=u.top+(u.height>>1)+r,d=Object.create(Object.prototype,{_items:{value:{}},effectAllowed:{value:"all",writable:!0},dropEffect:{value:"move",writable:!0},files:{get:function(){return this._items.Files}},types:{get:function(){return Object.keys(this._items)}},setData:{value:function(e,t){this._items[e]=t}},getData:{value:function(e){return this._items[e]}},clearData:{value:function(e){delete this._items[e]}},setDragImage:{value:function(e){}}});if(n=a.elementFromPoint(s,f),!n)throw new Error("The target element is not interactable and need to be scrolled into the view.");u=n.getBoundingClientRect(),e(t,"dragstart",101,function(){var i=n.getBoundingClientRect();c=i.left+s-u.left,g=i.top+f-u.top,e(n,"dragenter",1,function(){e(n,"dragover",101,function(){n=a.elementFromPoint(c,g),e(n,"drop",1,function(){e(t,"dragend",1)})})})})';
```

Comment:
wtf? :DD

Reply:
actually it is the only way to get working drag'n'drop :D
native protractor methods aren't working